### PR TITLE
Support for Kube 1.32

### PR DESCRIPTION
**Pull request by witomlin**
# CSA
## Added
- Support for Kube 1.32.
  - Container resizes now performed through `resize` subresource.
  - Ability to register an additional CA certificate (or chain) when building the kind node image for integration tests and sandbox scripts.

## Changed
- Upgrades Go to 1.23.3.
- Upgrades all dependencies.
- Renames controller-runtime controller name to shorten.

## Removed
- Examination of `AllocatedResources` within container status.
  - Not required and now behind feature gate in Kube 1.32.
- Controller name label from CSA metrics.

## Fixed
- Inconsistent status updates through informer cache race.
- CSA metrics not being published.

# Chart
## Added
- Support for Kube 1.32.
  - Cluster role includes `pods/resize` rule.

## Changed
- CSA version.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,8 +44,10 @@ non-production Kubernetes clusters.
     * [Reconciler](#reconciler)
     * [Scale](#scale)
     * [Kube API Retry](#kube-api-retry)
+    * [Informer Cache](#informer-cache)
```
Comment (nikos912000) [inline README.md:47]: This is getting a bit long for the README file. Would it be helpful for the users to have a separate `docs` folder with markdown files for different sections? Similar to [this](https://github.com/kubernetes/autoscaler/tree/master/vertical-pod-autoscaler/docs).

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -29,7 +29,7 @@ jobs:
     strategy:
       matrix:
         arg:
-          - '1.31'
+          - '1.32'
```
Comment (nikos912000) [inline .github/workflows/tests.yaml:32]: Don't we want to test against multiple versions for compatibility?

File: internal/controller/controller.go [hunk 3]
```diff
@@ -107,20 +107,22 @@ func (c *controller) Initialize(runtimeController ...runtimecontroller.Controlle
...
+			},
+		),
 	); err != nil {
 		return common.WrapErrorf(err, "unable to watch pods")
 	}
 
-	csametrics.RegisterAllMetrics(metrics.Registry, Name)
-	defer csametrics.UnregisterAllMetrics(metrics.Registry)
+	csametrics.RegisterAllMetrics(metrics.Registry)
 	c.initialized = true
```
Comment (nikos912000) [inline internal/controller/controller.go:126]: Isn't this flag prone to synchronisation issues where [Once.Do](https://pkg.go.dev/sync#Once.Do) would be safer?

File: .github/workflows/tests.yaml (hunk 2 above)
Comment (witomlin) [reply .github/workflows/tests.yaml:32]: We do but Kube 1.32 changes how containers are resized (uses the `resize` subresource) so tests can no longer run against older versions of Kube. Older versions of CSA have already been certified against Kube versions that it supports.

File: internal/controller/controller.go (hunk 3 above)
Comment (witomlin) [reply internal/controller/controller.go:126]: In theory, yes. But at the moment this containing function is only ever invoked once by `main.go`. Will change to use `Once.Do` though!

File: README.md (hunk 1 above)
Comment (witomlin) [reply README.md:47]: Yeah, good call. Could you create an issue for this please? We can implement separately.

File: internal/controller/controller.go (hunk 3 above)
Comment (witomlin) [reply internal/controller/controller.go]: https://github.com/ExpediaGroup/container-startup-autoscaler/pull/14/commits/a3c8cc94790eea83db9585b22f7028518fd51eea

**Review by sasidharsmit — APPROVED** [review]

**Pull request merged**